### PR TITLE
Migrate CLI to Bun

### LINKED ISSUE #363: CLI: Migrate to Bun
Refer #273

### PR DESCRIPTION
**Pull request by RanitMukherjee**
## Description

_As of now, only have modified the package.json of the CLI_ <br />

Fixes #363 

## Future Improvements

_This is surely incomplete as of now & the only motivation behind such a low effort PR is to understand the requirements better & actually get the work done._

## Mentions

 @rajdip-b 

## Developer's checklist

- [ ] My PR follows the style guidelines of this project
- [ ] I have performed a self-check on my work

**If changes are made in the code:**

- [ ] I have followed the [coding guidelines](https://google.github.io/styleguide/jsguide.html)
- [ ] My changes in code generate no new warnings
- [ ] My changes are breaking another fix/feature of the project
- [ ] I have added test cases to show that my feature works
- [ ] I have added relevant screenshots in my PR
- [ ] There are no UI/UX issues

### Documentation Update

- [ ] This PR requires an update to the documentation at [docs.keyshade.xyz](https://docs.keyshade.xyz)
- [ ] I have made the necessary updates to the documentation, or no documentation changes are required.

___

### **PR Type**
enhancement, configuration changes

___

### **Description**
- Added new scripts to `package.json` for Bun build, start, and dev commands.
- Modified the existing `dev` script to include Bun commands.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Configuration changes</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>package.json</strong><dd><code>Add Bun support to CLI scripts in package.json</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/cli/package.json

<li>Added new scripts for Bun build, start, and dev commands.<br> <li> Modified existing dev script to include Bun commands.<br>

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/400/files#diff-1625a7848c01883721bcd146381e2a6bff3c03814d4586559265d8e54a0c9565">+5/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Comment by rajdip-b** [comment]
It's true that the installation will be a pain initially. We will have the docs up for it. 

I think we can't migrate the API to bun because NestJS isnt supported. So, we are mostly aiming for the API Client and CLI to be migrated. And then if possible, we would migrate the frontend.

File: apps/cli/package.json
Comment (rajdip-b) [inline apps/cli/package.json:1]: Maybe you can swap out the existing commands with the new ones and see how it goes?

**Review by rajdip-b — CHANGES_REQUESTED** [review]

**Comment by RanitMukherjee** [comment]
> It's true that the installation will be a pain initially. We will have the docs up for it.
> 
> I think we can't migrate the API to bun because NestJS isnt supported. So, we are mostly aiming for the API Client and CLI to be migrated. And then if possible, we would migrate the frontend.

I just checked out Bun Issue in regard to NestJS compatability, and it seems as of 21st June 2024, most of support for NestJS has been established & separate issues regarding edge cases/specific troubleshooting are to be created. Can you clarify in regard to details in regard to specific issues we been encountering, maybe a issue can be raised against the same & once sorted, a full migration to Bun might not be far, thus allowing a simpler set up experience as well.

**Comment by rajdip-b** [comment]
We won't know for sure what exactly the blockers might be, until and unless someone starts working on it. That's why, we have parked it for the very end.

**Comment by Jarred-Sumner** [comment]
let us know if you run into any issues with nestjs in bun

**Comment by rajdip-b** [comment]
@Jarred-Sumner thanks! We are tracking our NestJS migration in #275. We will surely hit you up in case we hit a dead end :D

**Comment by rajdip-b** [comment]
We are closing this PR atm. We would like to stall it for later, and probably let our internal team handle the feature since it's a bit complex. Thanks for your efforts though @RanitMukherjee!

**Pull request closed without merging**